### PR TITLE
Porting from ACS to AKS

### PR DESCRIPTION
**Pull request by ppathan**
# Description and Motivation 
* ARM template change to deploy AKS
* Code change to get user credentials through SDK
* Work around for moving the public ip resource into AKS deployed RG to mitigate permissions issue
* Merging the kube config with local version if present to allow multiple kubectl configs

# Change type 

- [ ] Bug fix
- [x] New feature
- [x] Enhancement
- [x] Breaking change (breaks backward compatibility)

**Checklist:**
This code requires documentation change that will come soon. But i wanted to share this change with customer to try it out.

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [x] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/pcs-cli/381)

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -9,22 +9,22 @@
...
-        "shelljs": "0.7.8",
-        "underscore": "1.8.3",
-        "websocket": "1.0.28"
+        "base-64": "^0.1.0",
+        "bluebird": "^3.3.5",
+        "byline": "^5.0.0",
+        "js-yaml": "^3.5.2",
+        "jsonpath": "^1.0.0",
+        "request": "^2.72.0",
+        "shelljs": "^0.7.8 ",
```
Comment (jillcary) [inline package-lock.json]: nit: extra whitespace

File: src/deploymentmanager.ts [hunk 2]
```diff
@@ -198,42 +202,53 @@ export class DeploymentManager implements IDeploymentManager {
...
+                    const client = new NetworkManagementClient(this._credentials, this._subscriptionId);
+                    const aksResourGroup: string = 
+                    `MC_${outputs.resourceGroup.value}_${aksClusterName}_${resourceGroup.location}`;
+                    const moveInfo: ResourceModels.ResourcesMoveInfo = {
+                        resources: [ outputs.publicIPResourceId.value ],
+                        targetResourceGroup: `/subscriptions/${this._subscriptionId}/resourceGroups/${aksResourGroup}`
+                    };
+                    return this._client.resources.moveResources(outputs.resourceGroup.value, moveInfo)
+                    .then( () => {
+                        deployUI.stop({ message: `Crede ntials downloaded to config: ${chalk.cyan(kubeConfigPath)}` });
```
Comment (jillcary) [inline src/deploymentmanager.ts]: nit: space in `Crede ntials`

File: src/deploymentmanager.ts [hunk 3]
```diff
@@ -198,42 +202,53 @@ export class DeploymentManager implements IDeploymentManager {
...
-                    config.MessagesEventHubConnectionString = outputs.messagesEventHubConnectionString.value;
-                    config.MessagesEventHubName = outputs.messagesEventHubName.value;
-                    config.TelemetryStorgeType = outputs.telemetryStorageType.value;
-                    config.TSIDataAccessFQDN = outputs.tsiDataAccessFQDN.value;
-                    const k8sMananger: IK8sManager = new K8sManager('default', kubeConfigPath, config);
-                    deployUI.start('Setting up Kubernetes');
-                    return k8sMananger.setupAll();
+                    const aksClusterName: string = outputs.containerServiceName.value;
+                    const client = new NetworkManagementClient(this._credentials, this._subscriptionId);
+                    const aksResourGroup: string =
```
Comment (zhang-hua) [inline src/deploymentmanager.ts]: typo 'aksResoureGroup'

File: src/deploymentmanager.ts [hunk 4]
```diff
@@ -198,42 +202,53 @@ export class DeploymentManager implements IDeploymentManager {
...
-                    return k8sMananger.setupAll();
+                    const aksClusterName: string = outputs.containerServiceName.value;
+                    const client = new NetworkManagementClient(this._credentials, this._subscriptionId);
+                    const aksResourGroup: string = 
+                    `MC_${outputs.resourceGroup.value}_${aksClusterName}_${resourceGroup.location}`;
+                    const moveInfo: ResourceModels.ResourcesMoveInfo = {
+                        resources: [ outputs.publicIPResourceId.value ],
+                        targetResourceGroup: `/subscriptions/${this._subscriptionId}/resourceGroups/${aksResourGroup}`
+                    };
+                    return this._client.resources.moveResources(outputs.resourceGroup.value, moveInfo)
```
Comment (zhang-hua) [inline src/deploymentmanager.ts]: will the resource group for public IP resource still be remained?

File: src/index.ts [hunk 5]
```diff
@@ -93,7 +93,6 @@ const program = new Command(packageJson.name)
     .option('-w, --websiteName <websiteName>', 'Name of the website, default is solution name')
     .option('-u, --username <username>', 'User name for the virtual machine that will be created as part of the solution')
     .option('-p, --password <password>', 'Password for the virtual machine that will be created as part of the solution')
-    .option('--sshFilePath <sshFilePath>', 'Path to the ssh file path that will be used by standard deployment')
```
Comment (zhang-hua) [inline src/index.ts]: remove same option in the launch settings under .vscode folder

File: solutions/remotemonitoring/armtemplates/standard.json [hunk 6]
```diff
@@ -1029,39 +1046,37 @@
...
+            "name": "[variables('aksName')]",
             "properties": {
-                "orchestratorProfile": {
-                    "orchestratorType": "[variables('orchestratorType')]"
-                },
-                "masterProfile": {
-                    "count": "[variables('masterCount')]",
-                    "dnsPrefix": "[variables('mastersEndpointDNSNamePrefix')]"
+                "kubernetesVersion": "[parameters('kubernetesVersion')]",
+                "enableRBAC": false,
```
Comment (sushilraje) [inline solutions/remotemonitoring/armtemplates/standard.json]: Same question for RBAC as above.

File: solutions/remotemonitoring/armtemplates/standard-static-map.json [hunk 7]
```diff
@@ -1003,39 +1020,37 @@
...
-                    {
-                        "name": "agentpools",
-                        "count": "[variables('agentCount')]",
-                        "vmSize": "[variables('agentVMSize')]",
-                        "dnsPrefix": "[variables('agentsEndpointDNSNamePrefix')]"
-                    }
+                {
+                    "name": "agentpool",
+                    "count": 2,
+                    "vmSize": "[parameters('agentVMSize')]",
```
[the remark this answers is not in the document]
Comment (ppathan) [reply solutions/remotemonitoring/armtemplates/standard-static-map.json]: Nope forgot to update it. Done.

File: solutions/remotemonitoring/armtemplates/standard-static-map.json [hunk 8]
```diff
@@ -1003,39 +1020,37 @@
...
+            "name": "[variables('aksName')]",
             "properties": {
-                "orchestratorProfile": {
-                    "orchestratorType": "[variables('orchestratorType')]"
-                },
-                "masterProfile": {
-                    "count": "[variables('masterCount')]",
-                    "dnsPrefix": "[variables('mastersEndpointDNSNamePrefix')]"
+                "kubernetesVersion": "[parameters('kubernetesVersion')]",
+                "enableRBAC": false,
```
[the remark this answers is not in the document]
Comment (ppathan) [reply solutions/remotemonitoring/armtemplates/standard-static-map.json]: Can you send me the link that shows how AKS uses RBAC?

File: src/index.ts (hunk 5 above)
Comment (ppathan) [reply src/index.ts]: Done

File: src/deploymentmanager.ts (hunk 4 above)
Comment (ppathan) [reply src/deploymentmanager.ts]: Added comment.

**Pull request merged**